src/core/analytics.js

```diff
@@ -851,11 +852,23 @@ class Analytics {
       // check for reserved keys and log
       checkReservedKeywords(rudderElement.message, type);
 
-      // if not specified at event level, All: true is default
-      const clientSuppliedIntegrations = rudderElement.message.integrations || { All: true };
+      let clientSuppliedIntegrations = rudderElement.message.integrations;
+
+      if (clientSuppliedIntegrations) {
+        // structure user supplied integrations object to rudder format
+        transformToRudderNames(clientSuppliedIntegrations);
+      } else if (
+        this.useIntegrationsInEvents &&
+        Object.keys(this.loadOnlyIntegrations).length > 0
```

Comment:
if this.loadOnlyIntegrations can be null or undefined lets also check for that before Object.keys.

also better to move th check logic into a constant like shouldUseGlobalIntegrationsConfigInEvents and use the value in the confitional instead of placing the logic directly in the conditional.

easier to read, changle logic and reuse value if needed

Reply:
We initialize this.loadOnlyIntegrations as {} and perform Object.assign on this. So it can never be null or undefined but still added the checking.